tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -276,6 +280,128 @@ public function test_delayed_dependent_with_blocking_dependency_not_enqueued( $s
 		$this->assertStringContainsString( $expected, $output, 'Only enqueued dependents should affect the eligible strategy.' );
 	}
 
+	/**
+	 * Data provider for test_has_only_delayed_dependents.
+	 *
+	 * @return array
+	 */
+	public function get_data_to_test_has_only_delayed_dependents() {
+		return array(
+			'no_dependents'                        => array(
+				'set_up'     => static function () {
+					wp_enqueue_script( 'foo', 'https://example.com/foo.js', array(), null, array( 'strategy' => 'defer' ) );
+					return 'foo';
+				},
+				'async_only' => false,
+				'expected'   => true,
+			),
+			'one_delayed_dependent'                => array(
+				'set_up'     => static function () {
+					wp_enqueue_script( 'foo', 'https://example.com/foo.js', array(), null, array( 'strategy' => 'defer' ) );
+					wp_enqueue_script( 'bar', 'https://example.com/bar.js', array( 'bar' ), null, array( 'strategy' => 'defer' ) );
```

Comment:
Is this meant to depend on foo?
```suggestion
					wp_enqueue_script( 'bar', 'https://example.com/bar.js', array( 'foo' ), null, array( 'strategy' => 'defer' ) );
```